controllers/admin/AdminImportController.php

```diff
@@ -1260,161 +1805,180 @@ public function categoryImport($offset = false, $limit = false, &$crossStepsVari
-                $category_to_create->active = 1;
-                $category_link_rewrite = Tools::link_rewrite($category_to_create->name[$id_lang]);
-                $category_to_create->link_rewrite = AdminImportController::createMultiLangField($category_link_rewrite);
-                $category_to_create->id_parent = Configuration::get('PS_HOME_CATEGORY'); // Default parent is home for unknown category to create
-
-                if (($field_error = $category_to_create->validateFields(UNFRIENDLY_ERROR, true)) === true &&
-                    ($lang_field_error = $category_to_create->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true &&
-                    !$validateOnly && // Do not move the position of this test. Only ->add() should not be triggered is !validateOnly. Previous tests should be always run.
-                    $category_to_create->add()) {
-                    $category->id_parent = $category_to_create->id;
+                $categoryToCreate = new Category();
+                $categoryToCreate->name = AdminImportController::createMultiLangField($category->parent);
+                $categoryToCreate->active = 1;
+                $categoryLinkRewrite = Tools::link_rewrite($categoryToCreate->name[$idLang]);
+                $categoryToCreate->link_rewrite = AdminImportController::createMultiLangField($categoryLinkRewrite);
+                // Default parent is home for unknown category to create :
+                $categoryToCreate->id_parent = Configuration::get('PS_HOME_CATEGORY');
+
+                // FIXME needing such a comment about "&& !$validateOnly" position is a code smell. Refacto needed.
+                if (($fieldError = $categoryToCreate->validateFields(UNFRIENDLY_ERROR, true)) === true
+                    && ($langFieldError = $categoryToCreate->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true
+                    && !$validateOnly // Do not move the position of this test. Only ->add() should not be triggered is !validateOnly. Previous tests should be always run.
+                    && $categoryToCreate->add()
+                ) {
+                    $category->id_parent = $categoryToCreate->id;
                 } else {
                     if (!$validateOnly) {
-                        $this->errors[] = sprintf(
-                            $this->trans('%1$s (ID: %2$s) cannot be saved', array(), 'Admin.Advparameters.Notification'),
-                            $category_to_create->name[$id_lang],
-                            (isset($category_to_create->id) && !empty($category_to_create->id))? $category_to_create->id : 'null'
-                        );
+                        $this->errors[] = $this->trans('%1$s (ID: %2$s) cannot be saved', array(
+                            $categoryToCreate->name[$idLang],
+                            !empty($categoryToCreate->id) ? $categoryToCreate->id : 'null',
```

Comment:
Same comment about translation placeholder. Please use a variable name surrounded by `%%` an add it in the second parameter.